### DISCUSSION TITLE
Foxglove compatible with vision_msgs?

### DISCUSSION THREAD

**Discussion by lukashsmannheim** [foxglove/community#889] [General]
I would like to visualize labels from the [view of delft dataset](https://github.com/tudelft-iv/view-of-delft-dataset) in foxglove. Specifically I was interested in using a Detection3D message from [vision_msgs.](http://docs.ros.org/en/lunar/api/vision_msgs/html/msg/Detection3D.html)

The labels have the following form:

> bicycle 1757 1 -0.5150583918601345 1692.8588 873.00977 1935.0 1064.7266 0.9959256326426174 0.4582897348611458 1.737482152677817 5.230204792744421 2.477337074657124 8.676091008791296 0.027439126666472635 1

Is it compatible? If no, do you have any recommendation which message type I should use?

**Comment by defunctzombie** [comment]
Foxglove does not support `vision_msgs` directly - however, you can still visualize them by using either a [message converter](https://docs.foxglove.dev/docs/visualization/extensions/guides/create-message-converter/) or a [user-script](https://docs.foxglove.dev/docs/visualization/panels/user-scripts/).

For either one you can output a SceneUpdate message that has a [SceneEntity](https://docs.foxglove.dev/docs/visualization/message-schemas/scene-entity/) with a [TextPrimitive](https://docs.foxglove.dev/docs/visualization/message-schemas/text-primitive/)
